src/include/miopen/inline_vector.hpp

```diff
@@ -66,21 +96,44 @@ class InlineVector
         std::copy(data.begin(), data.end(), storage.begin());
     }
 
-    template <typename InputIterator>
-    InlineVector(InputIterator first, InputIterator last) : real_size(std::distance(first, last))
+    InlineVector(std::initializer_list<T>&& data) : real_size(data.size())
     {
         if(real_size > N)
         {
             MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
         }
 
-        std::copy(first, last, storage.begin());
+        std::copy(data.begin(), data.end(), storage.begin());
+    }
+
+    template <typename InputIterator>
+    InlineVector(InputIterator first, InputIterator last)
+    {
+        if constexpr(std::is_integral<InputIterator>::value)
+        {
+            InlineVector(size_t(first), T(last));
+        }
+        else
+        {
+            real_size = std::distance(first, last);
+            if(real_size > N)
+            {
+                MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
+            }
+            std::copy(first, last, storage.begin());
+        }
     }
 
     // Copy/move operator
     InlineVector& operator=(const InlineVector& inline_vec) = default;
     InlineVector& operator=(InlineVector&& inline_vec) noexcept = default;
 
+    // Compare operators
+    bool operator==(const InlineVector& riv) const { return storage == riv.storage; }
+    bool operator!=(const InlineVector& riv) const { return storage != riv.storage; }
+    bool operator<(const InlineVector& riv) const { return storage < riv.storage; }
+    bool operator>(const InlineVector& riv) const { return storage > riv.storage; }
```

Comment:
It must take `real_size` into account, since the rest of the array may not be initialized.
So you have to implement those operators instead of reusing it from `std::array`.